Automattic/wp-calypso
# Reduxify: lib/shortcodes, take two

The reapplication of #8857, with a slightly simpler implementation using the new action proxy.


To test:

* Checkout this branch
* Verify there are no regressions within:
  * Post editor shortcodes (rendering and editing items, captions, inserting, reordering and deleting items)
  * Post editor media modal (editing media item data, including title and secondary fields)
  * Post editor featured image (rendering and editing)
  * Media library - dropping a file, uploading a file through URL, uploading a file directly
* Verify all tests pass
* Verify nothing imports `lib/shortcodes`
